### DISCUSSION TITLE
Should we reorganize the repo ?

### DISCUSSION THREAD

**Discussion by guiv42** [helge17/tuxguitar#165] [General]
I see two major drawbacks of the historical structure of the repository, with a flat organization of about 70 folders at root level:
- from developer's perspective it's not very convenient. Basically it makes manual browsing of the code quite inefficient.
- more critical I think: from visitor's perspective it does not make this repo very readable. When reaching repo's main page, visitor needs to scroll to the bottom of page to see the readme and what this repo is about. For seomeone already aware of TuxGuitar and motivated this might be ok, but for a newcomer it's not very welcoming I think.

I can see several options to improve this.

1. reorganize code, adding a layer of folders
Targeting about 5-7 folders at root level. Something like Android/Desktop/Common/Test/Legacy might ease navigation in the code. Easy to say, but I must admit I am not yet capable of doing this sorting activity, since I don't have an exhaustive understanding of all folders (by far!).

2. If the impact is too significant (e.g. dependencies between folders, pom files, build script), then at least I suggest to create 1 single top-level folder "code" or "source", and to hide everything in that folder. Keeping only "misc", "readme", "license",... and a few files at top-level (the most relevant ones for a visitor).

3. Add a screenshot to "readme". The idea is to inform efficiently a visitor: "yes, you're at the right place, TuxGuitar is here", with a synthetic preview so that newcomers understand fast what it is about

4. I suggest moving "license, changes, install" files back at root level, so that a visitor can quickly access this useful content

These are just suggestions, any idea welcome.
I can spend some time on this activity if needed.

**Comment by helge17** [comment]
Your suggestions make sense, I can adapt the build script to the changes.

About [10 directories](https://github.com/helge17/tuxguitar/blob/master/misc/build_tuxguitar_from_source.sh#L153) are currently not used for building, probably they are outdated somehow.

The AUTHORS, CHANGES etc files were out of sync some time ago and I wanted to avoid redundancy, so I replaced the copies with soft links. But I agree that it's better to have the files in the root directory and the links in the subdirectories; commit a5cb02b259190eadace40c45817409ae82725144 swaps the files and links accordingly.

**Comment by guiv42** [comment]
OK, thanks.
So I will try to do something. I think the only things I won't be able to test are the full build script and the macOS build.
> About [10 directories](https://github.com/helge17/tuxguitar/blob/master/misc/build_tuxguitar_from_source.sh#L153) are currently not used for building

Edit: about "TuxGuitar-ui-toolkit-qt5", shall we consider the Qt version is abandoned?
I would vote for this, as I already spent much time to try to build/run this version without success (#20). In this case the corresponding sub-folder of "build-scripts" could also be deprecated

**Comment by helge17** [comment]
Qt would be very nice, but I don't see any progress on this in the near future either. So, yes, we should depreciate it.

**Comment by guiv42** [comment]
See proposal in PR #166 
Moving folders around was simpler than initially anticipated, I only needed to modify the pom files + one gradle file for android.
Symlinks to doc also needed to be updated, I had not anticipated that. Globally, the evolution is simple but very wide... so I just hope I haven't forgotten anything.

See PR description, I performed a basic test on all configurations I can normally run.  No detailed tests, I just checked TG starts and produces some sound when notes are added to the default song.

I did not analyze the impact on the global build script, I hope the adaptation will not be too complex. I also did not address 3rd point of my initial post (adding a screenshot in readme).

**Comment by guiv42** [comment]
Thanks for merging #166.
I had a quick look at the global build script, it seems to me that simply replacing `build-scripts/` by either `desktop/build-scripts/` or `android/build-scripts/` everywhere should do most of the job.

**Comment by guiv42** [comment]
Thanks for the build script update!
Closed.
